circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/replica/Replica.java

```diff
@@ -81,12 +82,12 @@
       ReplicaTableFactory replicaTableFactory,
       HousekeepingListener housekeepingListener,
       ReplicaCatalogListener replicaCatalogListener,
-      ReplicationMode replicationMode) {
+      TableReplication tableReplication) {
     super(replicaCatalog.getName(), replicaHiveConf, replicaMetaStoreClientSupplier);
     this.replicaCatalogListener = replicaCatalogListener;
     tableFactory = replicaTableFactory;
     this.housekeepingListener = housekeepingListener;
-    this.replicationMode = replicationMode;
+    this.tableReplication = tableReplication;
```

Comment:
NOABD: maybe nicer of you keep the field and do `replicationMode = tableReplicaion.getReplicationMode()` in the constructor once.

Reply:
Agreed, done